base/compiler/ssair/inlining.jl

```diff
@@ -1321,26 +1418,30 @@ end
 
 function handle_match!(
     match::MethodMatch, argtypes::Vector{Any}, flag::UInt8, state::InliningState,
-    cases::Vector{InliningCase}, allow_abstract::Bool = false)
+    cases::Vector{InliningCase}, allow_abstract::Bool = false, check_sparams::Bool = false)
```

Comment:
Now it seems better to change the `allow_abstract` and `check_sparams` to required positional arguments or even required keyword ones, and make their configuration explicit at callsites (if they're positional, we should use `#=allow_abstract=#` inline comment also). It would make future refactoring easier and safer.

Reply:
(Although now I want to make it required argument, ) the default value of `check_sparams` should be `true` to keep the inlining algorithm safe.